### PR TITLE
Added option to fade in new controller

### PR DESCRIPTION
**Pull request by designatednerd**
Love this concept. For the app we're using this in, our UX team preferred a fade-in action to the slide-over action, so I added it as a dev-settable option where the default action is still your original animation. Please let me know if you've got any questions, and thanks for putting this together!

### CODE REVIEW

File: TWTSideMenuViewController/TWTSideMenuViewController.m [hunk 1]
```diff
@@ -306,10 +306,35 @@ - (void)setMainViewController:(UIViewController *)mainViewController animated:(B
     [self.containerView addSubview:incomingViewController.view];
 
     incomingViewController.view.frame = self.containerView.bounds;
-    incomingViewController.view.transform = CGAffineTransformTranslate(incomingViewController.view.transform, outgoingStartX, 0.0f);
+
```
Comment (jnjosh) [inline TWTSideMenuViewController/TWTSideMenuViewController.m:309]: Would you mind cleaning up the extra space here?

File: TWTSideMenuViewController/TWTSideMenuViewController.m [hunk 2]
```diff
@@ -306,10 +306,35 @@ - (void)setMainViewController:(UIViewController *)mainViewController animated:(B
...
+    UIViewAnimationOptions options = UIViewAnimationOptionCurveEaseInOut;
+    switch (self.animationType) {
+        case TWTSideMenuAnimationTypeSlideOver: {
+            CGFloat outgoingStartX = CGRectGetMaxX(outgoingViewController.view.frame);
+
+            incomingViewController.view.transform = CGAffineTransformTranslate(incomingViewController.view.transform, outgoingStartX, 0.0f);
+            break;
+        }
+        case TWTSideMenuAnimationTypeFadeIn:
+            incomingViewController.view.alpha = .6;
```
Comment (jnjosh) [inline TWTSideMenuViewController/TWTSideMenuViewController.m]: Just for consistency, could you change this to 0.6f?

File: TWTSideMenuViewController/TWTSideMenuViewController.m [hunk 3]
```diff
@@ -306,10 +306,35 @@ - (void)setMainViewController:(UIViewController *)mainViewController animated:(B
...
+
     
     void (^swapChangeBlock)(void) = ^{
-        incomingViewController.view.transform = CGAffineTransformIdentity;
+        switch (self.animationType) {
+            case TWTSideMenuAnimationTypeSlideOver:
+                incomingViewController.view.transform = CGAffineTransformIdentity;
+                break;
+            case TWTSideMenuAnimationTypeFadeIn:
+                incomingViewController.view.alpha = 1;
```
Comment (jnjosh) [inline TWTSideMenuViewController/TWTSideMenuViewController.m]: Same here for consistency, 1.0f.

**Comment by jnjosh** [comment]
Looks good. Just a couple cleanup comments.

**Comment by designatednerd** [comment]
Requested changes have been made. Please let me know if you have any further questions. Thanks!

**Comment by jnjosh** [comment]
:+1:

**Pull request merged**